### PR TITLE
Added a map of services to serviceregistrations (not tested)

### PR DESCRIPTION
**Pull request by dancompton**
This isn't tested

### CODE REVIEW

File: portmapper.go [hunk 1]
```diff
@@ -3,9 +3,10 @@ package portmapper
 import (
 	"encoding/json"
 	"fmt"
-	"os"
-
 	"github.com/coreos/go-etcd/etcd"
+	"os"
+	"sync"
+	"time"
```
Comment (grepory) [inline portmapper.go:9]: run go imports and go fmt

File: portmapper.go [hunk 2]
```diff
@@ -84,29 +93,53 @@ func Unregister(name string, port int) error {
...
-
-	svc := &Service{name, port, os.Getenv("HOSTNAME")}
-	if err := svc.validate(); err != nil {
-		return err
-	}
+// Loops over each service type in the map and attempt to register
+func RegisterServices() {
+	for {
+		wg := &sync.WaitGroup{}
+		for service_name := range ServiceMap {
```
Comment (grepory) [inline portmapper.go:100]: go lint would catch this.

go prefers camel case, serviceName

File: portmapper.go [hunk 3]
```diff
@@ -84,29 +93,53 @@ func Unregister(name string, port int) error {
...
+				ServiceMap[service_name].Timestamp = time.Now().Unix()
+				ServiceMap[service_name].err = nil
+
+				if err := svc.validate(); err != nil {
+					ServiceMap[service_name].err = err
+				}
+
+				bytes, err := svc.Marshal()
+				if err != nil {
+					ServiceMap[service_name].err = err
```
Comment (grepory) [inline portmapper.go:116]: This just looks weird to me.

What is the intention of storing the error?

Comment (dancompton) [inline portmapper.go:116]: We don't need them?

Comment (grepory) [inline portmapper.go:116]: Probably not? Just log them for now. We can yeller them later.

**Pull request merged**